latch_cli/extras/nextflow/tasks/process.py

```diff
@@ -230,13 +193,102 @@ def get_fn_code(self, nf_script_path_in_container: Path):
                         "LATCH_EXPRESSION": {repr(self.statement)},
                         "LATCH_RETURN": {repr(json.dumps(self.ret))},
                         "LATCH_PARAM_VALS": json.dumps(channel_vals),
+                        "LATCH_PRE_EXECUTE": '{repr(pre_execute)}',
                     }},
                     check=True,
                 )
             except subprocess.CalledProcessError:
                 log = Path("/root/.nextflow.log").read_text()
                 print("\n\n\n\n\n" + log)
                 raise
+            """,
+            0,
+        )
+
+        return code_block
+
+    def get_fn_interface(self, nf_script_path_in_container: Path):
+        input_name, input_t = list(self._python_inputs.items())[0]
+        output_t = list(self._python_outputs.values())[0]
+
+        code_block = dedent(f"""\
+            def _read_resources() -> Dict:
+                try:
+                    with open(".latch/resources.json") as f:
```

Comment:
This isn't a super helpful comment but leads to two conclusions:
- We should introduce this overhead only when needed and the process is truly dynamic (creating rules to categorize process resource directives as dynamic by presence of closure or similar idea)
- We need to improve the scheduing + container creation for all of the little nodes not doing real compute (extends to generator, conditional evaluation, even operator nodes in addition to the pre-execute node) We can probably get away with not pulling our full base image and haveing some dedicated computer that schedules all of the little guys that lives for the full duration of the workflow. This could be the same computer that also hosts the runtime for the dynamic rewrite

Reply:
Yeah, I think the issue is that we don't know if it is truly dynamic until runtime because as you mentioned, the nextflow.config can be dynamic. The only case where this works is when the process itself has all cpu/mem/disk defined explicitly and they are not closures. Not sure how frequently this happens tho, looking the broken strings code it seems to be mostly defined in the config files. 
We could also require nextflow.config to be static at registration time, that seems like a reasonable limitation 

For scheduling in the short term, we could have a dedicated node group for all the small tasks where the node has the nextflow minimal base image already downloaded. This should make container creation and scheduling relatively quick